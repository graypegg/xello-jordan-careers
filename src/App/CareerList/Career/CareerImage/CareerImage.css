.CareerImage__image {
  position: relative;
  height: 350px;
  background-size: cover;
  background-position: center center;
}

.CareerImage__modeToggleButton,
.CareerImage__modeToggleButton--active {
  position: absolute;
  bottom: 10px;
  right: 10px;
  display: inline-block;
  padding: 10px 15px;
  color: #FFF;
  border-radius: 15px;
  cursor: pointer;
}

.CareerImage__modeToggleButton {
  background: var(--primary-dark);
}

.CareerImage__modeToggleButton--active {
  background: var(--dark);
}

.CareerImage__modeToggleButton--active:hover {
  background: var(--darker);
}

.CareerImage__modeToggleButton:hover {
  background: var(--primary-light);
}