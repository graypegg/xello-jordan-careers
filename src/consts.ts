export const enum ImageMode {
  High,
  Low
} 