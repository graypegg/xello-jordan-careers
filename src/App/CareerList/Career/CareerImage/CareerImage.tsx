import { ImageMode } from '../../../../consts'

import * as React from 'react'

import './CareerImage.css'

interface ICareerImageProps {
  image: string
}

interface ICareerImageState {
  mode: ImageMode
}

class CareerImage extends React.Component<ICareerImageProps, ICareerImageState> {
  constructor (props: ICareerImageProps) {
    super(props)

    this.state = {
      mode: ImageMode.Low
    }

    this.scaleImage = this.scaleImage.bind(this)
    this.toggleMode = this.toggleMode.bind(this)
  }

  public scaleImage (image: string) {
    return `${image}${this.state.mode === ImageMode.High ? '' : '?quality=70&maxWidth=500'}`
  }

  public toggleMode (): void {
    this.setState({
      mode: this.state.mode === ImageMode.High ? ImageMode.Low : ImageMode.High
    })
  }

  public render (): JSX.Element {
    return (
      <div className="CareerImage__wrapper">
        <div className="CareerImage__image" style={ { backgroundImage: `url(${this.scaleImage(this.props.image)})` } }>
          <div
            className={this.state.mode === ImageMode.High ? 'CareerImage__modeToggleButton--active' : 'CareerImage__modeToggleButton'}
            onClick={this.toggleMode}>
              {
                this.state.mode === ImageMode.High
                ? 'Show Low Res'
                : 'Show High Res'
              }
          </div>
        </div>
      </div>
    )
  }
}

export default CareerImage