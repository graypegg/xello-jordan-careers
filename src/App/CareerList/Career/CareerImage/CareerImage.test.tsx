
import * as React from 'react'
import { shallow, render } from 'enzyme'
import { ImageMode } from '../../../../consts'

import CareerImage from './CareerImage'

it('renders without crashing', () => {
  shallow(<CareerImage image="" />)
})

it('renders a low-res image', () => {
  const wrapper = render(<CareerImage image="test" />)
  expect(wrapper.find('.CareerImage__image').attr('style')).toBe(`background-image:url(test?quality=70&maxWidth=500)`)
})

it('renders a high-res image', () => {
  const wrapper = shallow(<CareerImage image="test" />)
  wrapper.find('.CareerImage__modeToggleButton').first().simulate('click')
  expect(wrapper.find('.CareerImage__image').props().style).toEqual({backgroundImage: `url(test)`})
})