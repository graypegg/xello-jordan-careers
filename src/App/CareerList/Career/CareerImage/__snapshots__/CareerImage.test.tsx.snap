// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CareerImage
    image=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="CareerImage__image"
        style={
          Object {
            "backgroundImage": "url(?quality=70&maxWidth=500)",
          }
        }
      >
        <div
          className="CareerImage__modeToggleButton"
          onClick={[Function]}
        >
          Show High Res
        </div>
      </div>,
      "className": "CareerImage__wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="CareerImage__modeToggleButton"
          onClick={[Function]}
        >
          Show High Res
        </div>,
        "className": "CareerImage__image",
        "style": Object {
          "backgroundImage": "url(?quality=70&maxWidth=500)",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show High Res",
          "className": "CareerImage__modeToggleButton",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Show High Res",
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="CareerImage__image"
          style={
            Object {
              "backgroundImage": "url(?quality=70&maxWidth=500)",
            }
          }
        >
          <div
            className="CareerImage__modeToggleButton"
            onClick={[Function]}
          >
            Show High Res
          </div>
        </div>,
        "className": "CareerImage__wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="CareerImage__modeToggleButton"
            onClick={[Function]}
          >
            Show High Res
          </div>,
          "className": "CareerImage__image",
          "style": Object {
            "backgroundImage": "url(?quality=70&maxWidth=500)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show High Res",
            "className": "CareerImage__modeToggleButton",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Show High Res",
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
